server/src/controllers/auth.controller.js

```diff
@@ -77,4 +82,48 @@ const signinUser = async (req, res) => {
   }
 };
 
-export { signupUser, signinUser };
+
+/**
+   * verify a new user
+   * @method verifyEmailToken
+   * @param {object} req
+   * @param {object} res
+   * @returns {(function|object)} Function next() or JSON object
+   */
+const verifyEmailToken = async (req, res) => {
+  const { email, token } = req.query;
+
+  const foundUser = await Users.findOne({ where: { email } });
+
+  if (!foundUser) return res.status(404).json({ status: 'failure', error: 'user does not exist' });
+
+  if (foundUser) {
+    if (foundUser.isVerified) return res.status(409).json({ status: 'failure', error: 'your email has been verified already' });
+
+    if (!foundUser.isVerified) {
+      const emailVerification = await EmailVerifications
+        .findOne({ where: { userId: foundUser.id, token } });
+
+      if (!emailVerification) return res.status(404).json({ status: 'failure', error: 'email verification failed' });
```

Comment:
I feel the error message here should be the same as when the token is expired. Like `Email verification failed. Your token might be expired or invalid`